kubelet/tests/test_kubelet.py

```diff
@@ -876,20 +920,172 @@ def test_kubelet_stats_summary_not_available(monkeypatch, aggregator, tagger):
     check._retrieve_stats.assert_called_once()
 
 
-def test_system_container_metrics(monkeypatch, aggregator, tagger):
+def test_process_stats_summary_not_source(monkeypatch, aggregator, tagger):
     check = KubeletCheck('kubelet', {}, [{}])
-    monkeypatch.setattr(
-        check, '_retrieve_stats', mock.Mock(return_value=json.loads(mock_from_file('stats_summary.json')))
+    pod_list_utils = PodListUtils(json.loads(mock_from_file('pods_windows.json')))
+    stats = json.loads(mock_from_file('stats_summary_windows.json'))
+
+    tagger.reset()
+    tagger.set_tags(WINDOWS_TAGS)
+
+    tags = ["instance:tag"]
+    check.process_stats_summary(pod_list_utils, stats, tags, False)
+    from pprint import pprint
+
+    pprint(aggregator._metrics)
+
+    # As we did not activate `use_stats_summary_as_source`, we only have ephemeral storage metrics
```

Comment:
but we do want to collect `*.runtime.*` and `*.kubelet.*` when `use_stats_summary_as_source` is deactivated don't we? we're already collecting these with the default config (in addition to `ephemeral_storage.usage`) and not collecting them will be a breaking change

Reply:
Yes but we don't have them on Windows, thus not showing in this test. You'll find them in the newly-added Linux test

Reply:
great, should we rename the method `test_process_stats_summary_not_source_windows` or update the comment to be more explicit and explain this?

Reply:
Updated the test case